dask/dataframe/io/sql.py

```diff
@@ -97,19 +90,39 @@ def read_sql_table(table, uri, index_col, divisions=None, npartitions=None,
     else:
         # function names get pandas auto-named
         kwargs['index_col'] = index_col.name
+
+    q = sql.select(columns).limit(5).select_from(table)
+    head = pd.read_sql(q, engine, **kwargs)
+
+    if divisions is None:
+        if limits is None:
+            # calculate max and min for given index
+            q = sql.select([sql.func.max(index), sql.func.min(index)]
+                           ).select_from(table)
+            minmax = pd.read_sql(q, engine)
+            maxi, mini = minmax.iloc[0]
+            dtype = minmax.dtypes['max_1']
+        else:
+            mini, maxi = limits
+            dtype = pd.Series(limits).dtype
+        if npartitions is None:
+            q = sql.select([sql.func.count(index)]).select_from(table)
+            count = pd.read_sql(q, engine)['count_1'][0]
+            bytes_per_row = (head.memory_usage(deep=True, index=True)).sum() / 5
+            npartitions = round(count * bytes_per_row / AUTO_BYTES_PER_CHUNK)
+        if dtype.kind == "M":
+            divisions = pd.date_range(
+                start=mini, end=maxi, freq='%iS' % (
+                    (maxi - mini) / npartitions).total_seconds()).tolist()
```

Comment:
My searching around gave this solution for linspace for times; I don't know when it would fail. I'll add the line, and expect it makes no difference (but should not hurt).